File: frontend/bulkApi/src/models/Bid.ts
import { DiscountScheme } from "./DiscountScheme";

export class Bid {
    bidId: number = 0;
    isInCart: boolean = false;
    quantity: number = 0;
    bidSuccessDate: Date | null = null;

    collectionAddress: string = "";

    discountSchemeId: number = 0;
    discountScheme: DiscountScheme | null = null;

    customerId: number = 0;

}

File: backend/BulkApi/Models/Bid.cs
using System;

namespace BulkApi.Models
{
    public class Bid
    {
        public int BidId { get; set; }
        
        //The InCart property is important
        public bool IsInCart { get; set; }

        public int Quantity { get; set; }
        public DateTime? BidSuccessDate { get; set; }
        public string CollectionAddress { get; set; }
        public double FinalDeliveryCharge { get; set; }

        //CustomerFK
        public string CustomerId { get; set; }
        public Customer Customer { get; set; }

        //DiscountSchemeFk
        public int DiscountSchemeId { get; set; }
        public DiscountScheme DiscountScheme { get; set; }

        
    }
}